sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/DynamicDestinations.java

```diff
@@ -132,6 +132,10 @@ void setSideInputAccessorFromProcessContext(DoFn<?, ?>.ProcessContext context) {
   /**
    * Returns an object that represents at a high level which table is being written to. May not
    * return null.
+   *
+   * <p>Returned object is expected to be unique for different destination tables involved over all
```

Comment:
Do users implement this method? If so, can we be more prescriptive: 

For example:

```
Returned object must be unique for different destination tables involved over all
The method must return a unique object for different destination tables involved over all

```

Reply:
thanks for the suggestion, yes, user provides implementation of this abstract class